Services/User/Profile/classes/class.ilPersonalProfileGUI.php

```diff
@@ -527,7 +527,61 @@ protected function showUserAgreement()
         $this->tpl->setPermanentLink('usr', null, 'agreement');
         $this->tpl->printToStdout();
     }
-    
+
+    protected function showConsentWithdrawalConfirmation() : void
+    {
+        $this->tabs->clearTargets();
+        $this->tabs->clearSubTabs();
+        $this->tpl->setTitle($this->lng->txt('withdraw_consent'));
+
+        $tpl = new \ilTemplate('tpl.withdraw_terms_of_service.html', true, true, 'Services/TermsOfService');
+        $tpl->setVariable('TERMS_OF_SERVICE_WITHDRAWAL_CONTENT', $this->lng->txt('withdraw_consent_info'));
+        $tpl->setVariable('FORM_ACTION', $this->ctrl->getFormAction($this));
+
+        $tpl->setVariable('WITHDRAW_TERMS_OF_SERVICE', $this->lng->txt('withdraw_usr_agreement'));
+        $tpl->setVariable('TXT_WITHDRAW', $this->lng->txt('withdraw'));
+        $tpl->setVariable('TXT_CANCEL', $this->lng->txt('cancel'));
+
+        $this->tpl->setContent($tpl->get());
+        $this->tpl->setPermanentLink('usr', null, 'agreement');
+        $this->tpl->printToStdout();
+    }
+
+    protected function cancelWithdrawal() : void
+    {
+        $this->user->writePref('consent_withdrawal_requested', 0);
+        ilInitialisation::redirectToStartingPage();
+    }
+
+    protected function withdrawAcceptance() : void
+    {
+        global $DIC;
+        $document = $this->termsOfServiceEvaluation->document();
+        $helper = new \ilTermsOfServiceHelper();
+        $helper->resetAcceptance($this->user, $document);
+
+        $shouldDeleteAccountOnWithdrawal = $DIC->settings()->get(
+            'tos_withdrawal_usr_deletion',
+            false
+        );
+        if($shouldDeleteAccountOnWithdrawal == 1 &&
```

Comment:
This is **not** correct. The `default` authentication mode could be `LDAP` (or something else). Please be aware that the `default mode concept` itself will be abandoned AFAIK. As long as this concept still exists, you'll have to check if the `default mode` (`ilSetting`) equals `'local'`.

Why do you try to delete only local accounts here?